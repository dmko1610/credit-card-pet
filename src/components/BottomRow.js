import React from "react";
import { View, Text, StyleSheet } from "react-native";

const CardholderName = (style) => {
  return <Text {...style}>John Wick</Text>;
};

const ExpireDate = (style) => {
  return <Text {...style}>04/21</Text>;
};

const ExpirationBlock = () => {
  const {
    commonText,
    expireDate,
    expirationBlockTitle,
    expirationBlockContainer,
  } = bottomRowStyle;
  return (
    <View style={expirationBlockContainer}>
      <Text style={StyleSheet.flatten([commonText, expirationBlockTitle])}>
        month/year
      </Text>
      <ExpireDate style={StyleSheet.flatten([commonText, expireDate])} />
    </View>
  );
};

const BottomRow = () => {
  const { commonText, cardholder, bottomRowContainer } = bottomRowStyle;
  return (
    <View style={bottomRowContainer}>
      <CardholderName style={StyleSheet.flatten([commonText, cardholder])} />
      <ExpirationBlock />
    </View>
  );
};

const bottomRowStyle = StyleSheet.create({
  bottomRowContainer: {
    flex: 1,
    flexDirection: "row",
    justifyContent: "space-between",
    margin: 10,
  },
  cardholder: {
    fontSize: 20,
    letterSpacing: 5,
    paddingBottom: 5,
    alignSelf: "flex-end",
  },
  expireDate: {
    fontSize: 17,
    letterSpacing: 2,
    paddingBottom: 5,
  },
  commonText: {
    color: "white",
  },
  expirationBlockTitle: {
    fontSize: 7,
    letterSpacing: 2,
  },
  expirationBlockContainer: {
    alignSelf: "flex-end",
  },
});

export default BottomRow;
