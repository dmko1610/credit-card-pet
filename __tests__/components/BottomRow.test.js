import React from "react";
import renderer from "react-test-renderer";

import BottomRow from "../../src/components/BottomRow";

describe("Bottom Row Creation", () => {
  it("Render component", () => {
    const tree = renderer.create(<BottomRow />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
