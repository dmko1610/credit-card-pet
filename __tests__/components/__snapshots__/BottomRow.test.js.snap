// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bottom Row Creation Render component 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "margin": 10,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "flex-end",
        "color": "white",
        "fontSize": 20,
        "letterSpacing": 5,
        "paddingBottom": 5,
      }
    }
  >
    John Wick
  </Text>
  <View
    style={
      Object {
        "alignSelf": "flex-end",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 7,
          "letterSpacing": 2,
        }
      }
    >
      month/year
    </Text>
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 17,
          "letterSpacing": 2,
          "paddingBottom": 5,
        }
      }
    >
      04/21
    </Text>
  </View>
</View>
`;
